TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -9,7 +9,20 @@ import Foundation
 
 class NetworkManager {
     
+    private var lastRequestTime: Date = .now
+    
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
+        if self.isTimeIntervalLongEnough(lastRequest: lastRequestTime) {
+            self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
+        } else {
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
+                self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
```

Comment:
DispatchQueue.global().asyncAfter must execute `get` method instead of  `executeNetworkGet` to check the timing again in 12.5 ms. (Because other request could be sent by this time)